presto-cache/src/main/java/com/facebook/presto/cache/LocalRangeCacheManager.java

```diff
@@ -345,6 +350,35 @@ private boolean write(FileReadRequest key, byte[] data, Path newFilePath)
         return true;
     }
 
+    private int copyFromFileInChunks(File newFile, LocalCacheFile sourceFile, long startPos)
+            throws IOException
+    {
+        int totalBytesRead = 0;
+        try (FileInputStream fileInputStream = new FileInputStream(new File(sourceFile.getPath().toUri()))) {
+            fileInputStream.getChannel().position(startPos);
+            int readBytes = 0;
+            byte[] buffer = threadLocal.get();
+
+            while ((readBytes = fileInputStream.read(buffer)) != -1) {
+                if (readBytes > 0 && !Files.exists(newFile.toPath())) {
+                    Files.createFile(newFile.toPath());
+                }
```

Comment:
Could you handle `readBytes == 0` case and write a unit test for it (if that case is even possible)?

Reply:
that case shouldn't be possible now given the ```>0``` condition.. 
Earlier, the only way this would have happened is if we passed a buffer of length 0, but since the buffer size is not being supplied from outside but the class itself, this shouldn't be an issue imo? We can chat more about it if you have concerns..